Automattic/wp-calypso
# Implement dedicated path to handle redirects to Titan control panel

#### Changes proposed in this Pull Request

* This PR implements a dedicated component to handle redirection from Calypso to the Titan control panel. When we want to open the control panel, we launch a new window to the Calypso URL for the site and domain, which then calls the server to generate a fresh URL, and redirects to it as soon as it can.

The component itself is set up _not_ to use the site selector and left nav, and shows only the Titan logo along with a "Redirecting..." message while waiting for the URL to be fetched.

A huge hat-tip to @eltongo for the simplified approach here. (My initial attempt here was really fragile.)

#### Testing instructions

* Ensure that the `titan/phase-2` feature flag is enabled and the `titan/iframe-control-panel` feature flag is disabled. (`?flags=titan/phase-2,-titan/iframe-control-panel`)
* Navigate to the email management page for a Titan/Email subscription (Manage -> Domains -> [your_domain] -> Manage your email accounts).
* Verify that clicking on the control panel link launches a Calypso page in a new tab/window, and that the new tab looks reasonable and redirects to Titan successfully and reasonably quickly.

It's probably worth testing a few times, as I discovered that we need to force fetching the state from the new tab/window. I could make this happen more regularly by running this in a local Calypso environment and loading the developer tools/console as soon as the tab opened -- Calypso intentionally simulates rehydrating state in that situation, so it's worth checking the console to make sure that this flow works even when that rehydration occurs. (I ended up needing to simplify the `componentDidUpdate()` logic to simply trigger the network request as soon as we have the `domain` object in hand, instead of waiting for the fetch to complete, as there is no guarantee the domain will be loaded by the time we get the state update for the fetch complete.)

#### Screenshot
<img width="1269" alt="control_panel_redirect_page" src="https://user-images.githubusercontent.com/3376401/105476986-f97ed780-5ca9-11eb-86cd-8209093bb7f7.png">
